shared/vendored/snakemake/versioning.smk

```diff
@@ -0,0 +1,163 @@
+            return
+
+        try:
+            spec = SpecifierSet(declared_spec)
+        except InvalidSpecifier:
+            # TODO XXX: I declared version '!31' which didn't raise an InvalidSpecifier but also didn't work as expected
+            # as installed version 31.3.0 was acceptable. '!31.*' or '!31.3' or '!31.3.0' failed.
+            self.declaration_errors.append(f"This pathogen declared an invalid version specification for {name!r} of {declared_spec}")
+            return
+        
+        ok = spec.contains(version)
+        # print(f"[DEBUG] Checking python dependency: {name!r} installed: {version} requirements: {spec} OK? {ok}", file=stderr)
+        if not ok:
+            self.version_incompatibilities.append(f"Python dependency {name!r} version incompatibility. You have {version} but this pathogen declares {spec}")
+
+    def check_cli_version(self, name: str, declared_spec: str) -> None:
+        """
+        Check whether the requested *name* is (a) installed and (b) reports a version
+        which satisfies the *declared_spec*. Both (a) and (b) are achieved by calling
+        `<name> --version`.Any errors or if the declared dependency isn't satisfied are
+        logged.
+        """
+        cmd = [name, "--version"]
+        try:
+            proc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=False)
+            output = ((proc.stdout or "") + " " + (proc.stderr or "")).strip()
+        except FileNotFoundError:
+            self.not_found_dependencies.append(f"CLI program {name!r} was not found. This pathogen requires a version satisfying {declared_spec}")
+            return
+
+        m = re.search(r"\d+(\.\d+(\.\d+)?)?([.-][0-9A-Za-z]+)*", output)
+        #                1   . 2   . 3           alpha etc
+        if not m:
+            self.unexpected_errors.append(f"CLI program {name!r} reported a version of {output} which we were unable to parse")
+            return
+
+        try:
+            version = Version(m.group(0))
+        except InvalidVersion: # <https://packaging.pypa.io/en/stable/version.html#packaging.version.InvalidVersion>
+            self.unexpected_errors.append(f"CLI program {name!r} reported a version of {m.group(0)} which we were unable to parse")
+
+        try:
+            spec = SpecifierSet(declared_spec)
+        except InvalidSpecifier:
+            self.declaration_errors.append(f"This pathogen declared an invalid version specification for CLI program {name!r} of {declared_spec}")
+            return
+
+        ok = spec.contains(version)
+        # print(f"[DEBUG] Checking CLI program: {name!r} installed: {version} requirements: {spec} OK? {ok}", file=stderr)
+        if not ok:
+            self.version_incompatibilities.append(f"CLI program {name!r} version incompatibility. You have {version} but this pathogen declares {spec}")
+
+
+def _read_nextstrain_pathogen_yaml(path: str) -> Dict:
```

Comment:
CI error:

```NameError in file /home/runner/work/zika/zika/phylogenetic/../shared/vendored/snakemake/versioning.smk, line 115:
name 'Dict' is not defined
```

Type hints with python are really frustrating because of the need to support different python versions. I'll just strip the hints from this file I think.